cockroachdb/cockroach
# ui: Cypress testing library integration

Depends on: https://github.com/cockroachdb/yarn-vendored/pull/18
Resolves: #46315

How to start tests: `yarn run cypress open`

- Add cypress library
- First e2e test on decommissioning node
and how Node overview list updates its state.

Integration of cypress library requires to setup
webpack, typescript and pre processing plugin to
make it work. All of this routine is isolated in
./cypress directory.

To interact with `roachprod`, Makefile is added as
abstraction level and a pool of available commands.
`./cypress/support/commands.ts` has custom Cypress
functions which execute make commands.

Cypress.get function is extended with custom logic
to provide custom log messages in Cypress logging
panel instead of logging CSS selectors.

And one more mention about test case, here is
main logic it validates:
- starts cluster with 4 nodes
- validates that all nodes are live
- decommission one node, and stop it
- check that 3 live and 1 node is Decommissioning
- wait while unavailable node is become decommissioned
- check that decommissioned node is shown in
Decommissioned nodes table

Release note: None

Release justification: non-production code changes
